Sources/PrivacyDashboard/BrokenSiteReporting/ReferrerInfo.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  ReferrerInfo.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import WebKit
+
+public struct ReferrerInfo {
+    @MainActor
+    public static func isSERPReferred(from webView: WKWebView) async -> Bool {
+        do {
+            if let result = try await webView.evaluateJavaScript("document.referrer") as? String {
+                return result.contains("duckduckgo.com")
+            }
+        } catch {
+            return false
+        }
+
+        return false
+    }
+}
```

Comment:
Id make it as an extension on WKWebView, like so:
public static async var isSERPReferred: Bool {}
and I'd actually rename it to better reflect what it does so:
public static async var isCurrentSiteReferredFromDuckDuckGo: Bool {}

then you can use it as webView.isCurrentSiteReferredFromDuckDuckGo